tasks/managed/push-rpm-to-koji/push-rpm-to-koji.yaml

```
@@ -299,6 +307,65 @@ spec:
         - mountPath: /etc/secret
           name: secret-volume
           readOnly: true
+    - name: push-build-to-artifact-storage
+      # The pipelineImage will be fetching from get-secrets task with koji installed there
+      image: images.paas.redhat.com/brew/rpmbuild-pipeline-environment-container:latest@sha256:82c22459ce51c59f05db59796d335599d42410282bc031e53eb2f419e5b8a606
+      computeResources:
+        limits:
+          memory: 256Mi
+        requests:
+          memory: 256Mi
+          cpu: 250m
+      env:
+        - name: SNAPSHOT_SPEC_FILE
+          value: $(params.dataDir)/$(params.snapshotPath)
+        - name: PIPELINE_METADATA
+          value: '$(params.releasePipelineMetadata)'
+        - name: DATA_FILE
+          value: $(params.dataDir)/$(params.dataPath)
+      script: |
+        #!/usr/bin/env bash
+        set -euxo pipefail
+
+        if [ ! -f "${SNAPSHOT_SPEC_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+        fi
+
+        if [ ! -f /etc/rok-access/cli.toml ] ; then
+            echo "No rok-access provided."
+            exit 1
+        fi
+
+        KOJI_IMPORT_DRAFT=$(jq -r '.pushOptions.koji_import_draft' "${DATA_FILE}")
+
+        if [[ "$KOJI_IMPORT_DRAFT" == "false" ]]; then
+            echo "Draft build skipping Artifact Storage"
+            exit 1
```

Should this cause pipeline failure?

```suggestion
            exit 0
```

If so, it would be better to skip the whole task:
- get `.pushOptions.koji_import_draft` value in `collect-task-params` task in the pipeline (`pipelines/managed/push-rpm-to-koji/push-rpm-to-koji.yaml`)
- pass the value to the task
- check the value (`when`) in this step

---

agree with @hluk regarding 'graceful skip' instead of error.
But I have another question: can you please clarify when the Pulp transfer should happen? during draft or final import?
It's a bit confusing, it looks like not so accurate/wrong message or wrong logic here ("$KOJI_IMPORT_DRAFT" should be == true)

---

Oh, true. The condition seems wrong based on the message. Also, `KOJI_IMPORT_DRAFT` (and `.pushOptions.koji_import_draft`) can be empty in which case it defaults to "import draft build". So the condition here could be `"$KOJI_IMPORT_DRAFT" != "false"` instead.

> But I have another question: can you please clarify when the Pulp transfer should happen? during draft or final import?

I believe we currently **only import draft builds** (during merge requests) and finally (after MR is merged) use `promote-koji-draft-build` pipeline task to **promote** the last imported draft build (based on the merged commit SHA).

---

I think this is supposed to happen during final import.